clusterInfo.py

```diff
@@ -247,3 +252,128 @@ def _match_name(name: str) -> bool:
         validate_cluster_info(cluster_info)
 
     return cluster_info
+
+
+###############################################################################
```

Comment:
What is this?

Reply:
it's a comment that acts as a visual separator between the code above and blow.

The code below is different because it's only relevant for the `if __name__ == "__main__":` case.

Reply:
Move tool related code into separate .py file and remove the separator.

Reply:
I can. But `clusterInfo.py` seems quite an apt name for this tool that fetches ... the cluster-info.

Also, many PY files have code like:
```
def main() -> None:
    pass

if __name__ == "__main__":
    main()
```

Most of the time `main()` is a NOP, sometimes it also does something useful. In any case, it seems very common that CDA's python modules are also directly executable programs.

I drop the `#####` line and rename some functions to have a `_main_` prefix (to make it clearer, that they are only used by `main()`.

If you still want to move out the code to another place, then please suggest a name for the new file.

Reply:
add it at the same level as deploy and snapshot

Reply:
I don't understand what you mean. Is there still a problem with the PR?

Reply:
You can run `cda.py deploy ...`, and `cda.py snaptshot` today. Can you add it so that it becomes possible to run `cda.py clusterinfo`?

Reply:
the command line API of `cda.py` is `./cda.py config {deploy,snapshot}`.

For `clusterinfo`, there is no `config`.

I don't think that the existing API can be sensibly extended in that way. The API could be broken, to move the verb first. But that seems out of scope for this branch.

Reply:
OK, I agree. This is a detail that we can tackle later.